packages/classic/src/payments.ts

```diff
@@ -0,0 +1,51 @@
+import { addPaymentHandler } from '@devvit/payments';
+import { type RedisClient } from '@devvit/public-api';
+import { DateTime } from 'luxon';
+
+class PaymentsRepo {
+  #redis: RedisClient;
+  constructor(redis: RedisClient) {
+    this.#redis = redis;
+  }
+  async addHardcoreModeLifetimeAccess(userId: string) {
+    await this.#redis.set(`hardcore-mode-lifetime-access:${userId}`, `1`);
+  }
+  async addHardcoreMode7DayAccess(userId: string) {
+    const sevenDaysFromNow = DateTime.now().plus({ days: 7 });
+    await this.#redis.set(`hardcore-mode-seven-day-access:${userId}`, sevenDaysFromNow.toISO());
+  }
```

Comment:
I don't think we should have these as separate keys. They are unnecessarily tightly coupled with the products and their timeframes.

I think there should be one `hardcore-mode-access` that contains a number representing the time the access is due to expire. If the number is -1, it is considered unlimited. This makes the access time much more flexible.